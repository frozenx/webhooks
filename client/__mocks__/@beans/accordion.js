

import React from 'react';

const Accordion = (props) => (<div {...props} />);

export default { Accordion };