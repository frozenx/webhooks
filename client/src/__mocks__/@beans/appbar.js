import React from 'react';
const AppBar = (props) => (<div {...props}/>);

export default AppBar;
