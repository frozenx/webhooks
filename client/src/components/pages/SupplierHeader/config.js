export const presets = {
    status: {
        depFields: ["partnerEndDateActive"],
        depPresets: {
            partnerEndDateActive: {
                depKey: 'showToggle',
                val: 'Inactive',
                op: '===',
                return: true,
                default: false
            }
        }
    },
    isPrimaryAddFlag: {
        depFields: ['addressName'],
        depPresets: {
            addressName: {
                depKey: 'required',
                val: 'Yes',
                op: '===',
                return: true,
                default: false
            }
        }
    },
    uuid: {
        depFields: ['expiredDate'],
        depPresets: {
            expiredDate: {
                depKey: 'required',
                val: 'uuid',
                op: '===',
                return: false,
                default: true
            }
        }
    },
    contactStatus: {
        depFields: ['inactiveDate'],
        depPresets: {
            inactiveDate: [{
                depKey: 'required',
                val: 'A',
                op: '===',
                return: false,
                default: true
            }, {
                depKey: 'regexRule',
                val: 'A',
                op: '===',
                return: "",
                default: ""
            }],
            administrativeContactFlag: {
                depKey: 'required',
                val: 'A',
                op: '===',
                return: false,
                default: true
            }
        }
    }
}