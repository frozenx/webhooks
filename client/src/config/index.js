const rootPath = '/tradingpartner';
const toolkitPath = 'https://toolkit.tesco.com';

module.exports = {
  rootPath,
  redirect: {
    pageNotFound: `${toolkitPath}/404`,
    unauthorized: `${toolkitPath}/sign-in/#/403`,
    serviceUnavailable: `${toolkitPath}/503`,
    serverError: `${toolkitPath}/500`,
    inactive: `${toolkitPath}/registration/#/set-details`,
    deactivated: `${toolkitPath}/sign-in/#/451`,
    b2b: '/toolkit/clubcard/',
    externalUrl: `${toolkitPath}/sign-in/`,
  },

  endPoints: {
    searchAndCreate: `${rootPath}/api/searchAndCreate`,
    supplier: `${rootPath}/api/supplier`,
    supplierHeaderEndPoint: `${rootPath}/api/supplier-header`,
    supplierSiteEndPoint: `${rootPath}/api/supplier-site`,
    paymentEndPoint: `${rootPath}/api/supplier-payments`,
    profileEndPoint: `${rootPath}/api/profile`,
    supplierPage: `${rootPath}/supplier`,
    logoutEndpoint: '/logout',
    uamEndPoint: `${rootPath}/api/uam`,
    siteGetDetails: `${rootPath}/api/site-details`,
    paymentGetDetails: `${rootPath}/api/payment-details`,
    getSupplierDetails: `${rootPath}/api/supplier-details`,
    getSiteDetails: `${rootPath}/api/supplier-details/sites`,
    getPaymentDetails: `${rootPath}/api/supplier-details/payments`,
    searchMetaData: `${rootPath}/api/search-meta-data`,
    getPartnerDetails: `${rootPath}/api/partner-details-search`,
    addressEndPoint: `${rootPath}/api/address`,
    saveAddress: editMode => editMode ? `${rootPath}/api/address?operation=patch` : `${rootPath}/api/address`,
    saveContact:editMode=>editMode? `${rootPath}/api/contact?operation=patch` : `${rootPath}/api/contact`,
    getAddressesPerPage: (pageNumber, partnerUuid) => `${rootPath}/api/adresses?partnerUuid=${partnerUuid}&pageNumber=${pageNumber}`,
    getContactsPerPage: (pageNumber, partnerUuid) => `${rootPath}/api/contacts?partnerUuid=${partnerUuid}&pageNumber=${pageNumber}`

  },

  regex: {
    nameRegex: /^[A-Za-z0-9\s]+$/,
    numberRegex: /^[0-9]*$/,
  },

  collectionRecords: ['address', 'contact'],
  addressKey: 'address',
  pageSize: 10,

  siteAddressList: [
    { id: 'primaryAddress', label: 'Primary address'},
    { id: 'orderingAddress', label: 'Ordering address'},
    { id: 'remmitanceAddress', label: 'Remmitance address'},
    { id: 'rfqBiddingAddress', label: 'RFQ bidding address'},
    { id: 'shippingAddress', label: 'Shipping address'},
  ],

  siteContactList: [
    { id: 'billToContact', label: 'BillTo Contact'},
    { id: 'shipToContact', label: 'ShipTo Contact'},
    { id: 'correspondenceContact', label: 'Correspondence Contact'},
    { id: 'dunningContact', label: 'Dunning Contact'},
    { id: 'InvoicesContact', label: 'Invoices Contact'},
  ],

  collectionKeys: {
    address: 'Back to Address book',
    contact: 'Back to Contact book',
    default: 'Back to Partner overview'
  },

  routerPath: {
    successfulCreationOfSupplier: `${rootPath}/supplier`,
  },

  entityTypes: {
    partner: 'partner',
    site: 'site',
    tnc: 'tnc',
  },
  forceCapitalizationAttributes: {
    partnerName: true,
  },
};
