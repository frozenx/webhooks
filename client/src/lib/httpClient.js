const redirectUrls = require('../config').redirect;


const handleError = (errorStatus) => {
    if (errorStatus === 403) {
        window.location = redirectUrls.unauthorized;
    } else if (errorStatus === 401) {
        window.location = redirectUrls.externalUrl;
    }
    else if (errorStatus === 404) {
        window.location = redirectUrls.pageNotFound;
    }
    else {
        window.location = redirectUrls.serverError;
    }
}

async function get(url) {
    try {
        const response = await fetch(url, {
            method: 'get',
            headers: {
                'Accept': 'application/json',
                'Content-Type': 'application/json',
            },
        });
        if (response.ok) {
            const body = await response.json();
            return body;
        } else {
            throw response;
        }
    }
    catch (err) {
        if (err.status) {
            handleError(err.status);
        }
    }
}

async function post(url, payload) {
    try {
        const response = await fetch(url, {
            method: 'POST',
            body: JSON.stringify(payload),
            headers: {
                'Accept': 'application/json',
                'Content-Type': 'application/json',
            },
        });
        if (response.ok) {
            const body = await response.json();
            return body;
        } else {
            throw response;
        }
    }
    catch (err) {
        if (err.status) {
            handleError(err.status);
        }
    }
}


export default { get, post };

