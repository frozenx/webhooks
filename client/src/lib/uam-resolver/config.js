import React from "react";
import CreateSupplier from "../../components/pages/CreateSupplier/CreateSupplier";
import ManageSupplier from "../../components/pages/CreateSupplier/ManageSupplier";
import Supplier from "../../components/pages/SupplierHeader";
import { Link } from 'react-router-dom';

const routesConfig = {
  tradingpartner: {
    exact: true,
    pathname: "/tradingpartner",
    component: ManageSupplier
  },
  "tradingpartner/manage": {
    exact: true,
    pathname: "/tradingpartner/manage",
    component: CreateSupplier
  },  
  "tradingpartner/supplier": {
    exact: true,
    pathname: "/tradingpartner/supplier",
    component: Supplier
  }
};

const navigationAccessList = ["tradingpartner", "tradingpartner/manage"];

const navigationConfig = {
  tradingpartner: {
    id: "createsupplier",
    hasPopup: false,
    altLink: () => (
      <Link to="/tradingpartner/manage" id="createsupplier">
        Create new partner
      </Link>
    ),
    text: "Create new supplier"
  },
  "tradingpartner/manage": {
    id: "managesupplier",
    hasPopup: false,
    altLink: () => (
      <Link to="/tradingpartner/" id="managesupplier">
        Manage partner
      </Link>
    ),
    text: "Manage suppliers"
  }
};

export default {
  routesConfig,
  navigationAccessList,
  navigationConfig
};
