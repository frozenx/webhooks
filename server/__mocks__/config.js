const getMock = jest.fn().mockReturnValue({});

module.exports = {
    get: getMock
}