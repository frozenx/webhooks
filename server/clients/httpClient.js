const axios = require('axios');
const logger = require('../lib/logger')(module);
const ErrorBuilder = require('../lib/error-handlers/ErrorBuilder');
const config = require('../config').get(process.env.APP_ENV);
const secrets = require('../secrets');

class HttpClient {
  constructor(accessToken) {
    this.accessToken = `Bearer ${accessToken}`;
  }

  /* eslint-disable */
  get(url, customHeaders = {}, params = {}) {
    return axios({
      url,
      params,
      headers: Object.assign({ Authorization: this.accessToken }, customHeaders),
    })
      .then(response => response.data)
      .catch((error) => {
        // logger.error({
        //   message: {
        //     logMsg: error.message,
        //     statusCode: error.status
        //   }
        // });
        // throw error;

        const httpError = new ErrorBuilder('http', error);
        throw httpError;
      });
  }
  /* eslint-enable */

  post(url, data = {}, customHeaders) {
    return axios({
      url,
      data,
      method: 'post',
      headers: { Authorization: this.accessToken, ...customHeaders },
    })
      .then(response => response.data)
      .catch((error) => {
        const httpError = new ErrorBuilder('http', error);
        throw httpError;
      });
  }

  put(url, data, customHeaders) {
    return axios({
      url,
      data,
      method: 'put',
      headers: { Authorization: this.accessToken, ...customHeaders },
    })
      .then(response => response.data)
      .catch((error) => {
        const httpError = new ErrorBuilder('http', error);
        throw httpError;
      });
  }
  getClientToken() {
    const url = config.identity.baseUrl + config.identity.getAccessToken;
    return axios({
      url,
      data: {
        'client_id': secrets.client_id, //to be added
        'grant_type': secrets.superUserForIdentityService.grant_type,
        'scope': secrets.superUserForIdentityService.scope,
        'username': secrets.superUserForIdentityService.username, //to be added
        'password': secrets.superUserForIdentityService.password //to be added
      },
      method: 'post',
      headers: { 'content-type': 'application/json' },
    })
      .then(response => response.data.access_token)
      .catch((error) => {
        const httpError = new ErrorBuilder('http', error);
        throw httpError;
      });
  }
}

module.exports = HttpClient;
