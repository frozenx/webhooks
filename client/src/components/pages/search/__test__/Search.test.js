jest.mock('../../../../lib/httpClient');

import React from 'react';
import Enzyme, { shallow } from 'enzyme';
import SearchSupplier from '../index';
import i18Data from '../../../../languagepack';
import renderer from 'react-test-renderer';
import { DefaultThemeProvider } from '@beans/theme';
import httpClient from '../../../../lib/httpClient';
import config from './confg';


describe('Given the Search component is rendered', () => {    
    let Wrapper,
        tree;
    const mockHistory = {push: jest.fn()}
    beforeEach(() => {
        window.fetch = jest.fn();
        httpClient.get.mockClear();
        tree = renderer.create(
            <DefaultThemeProvider>
                <SearchSupplier history={mockHistory}/>
            </DefaultThemeProvider>).toJSON()
        Wrapper = shallow(<SearchSupplier history={mockHistory}/>)
    })
    test('it should create a snapshot', () => {        
        expect(tree).toMatchSnapshot();
    });
    test('it should not have any error when component is rendered initially', () => {
    expect(Wrapper.find(".input-error")).toHaveLength(0);        
    });

    describe('Given the input value has changed', () => {        
        test('it should change the value of Supplier name for onChange', () => {        
            const event = {target: {id: 'name', value: 'Uniliver'}}            
            Wrapper.find('.name').simulate('change', event);                                    
            expect(Wrapper.state().inputs['name']).toBe('Uniliver')
        });
        test('it should change the value of Supplier number for onChange', () => {                    
            const event = {target: {id: 'number', value: '12345'}}        
            Wrapper.find('.number').simulate('change', event);        
            expect(Wrapper.state().inputs['number']).toBe('12345');
        });
        test('it should reset the error if it already has', () => {                    
            Wrapper.setState({error:{name: i18Data.invalidSupplierName, number: i18Data.invalidSupplierNumber} })
            const event = {target: {id: 'number', value: '12345'}}        
            Wrapper.find('.number').simulate('change', event);        
            expect(Wrapper.state().error['name']).toBe('');
            expect(Wrapper.state().error['name']).toBe('');
        });
        test('it should reset the disabled fields', () => {                    
            const event = {target: {id: 'name', value: ''}}        
            Wrapper.find('.number').simulate('change', event);
            Wrapper.setState({inputs:{name: '', number: ''} })        
            expect(Wrapper.state().disabled['name']).toBe(false);
            expect(Wrapper.state().disabled['number']).toBe(false);
        });
    });

    describe('Given for onBlur', () => {
        test('it should show error if name fails the valitation', () => {        
            const event = { target: {id: 'name', value:'uni 7-'}}
            Wrapper.find('.name').simulate('blur', event);
            expect(Wrapper.state().error['name']).toBe(i18Data.invalidSupplierName);
        });
        test('it should error if number fails the valitation', () => {        
            const event = { target: {id: 'number', value:'abc'}}
            Wrapper.find('.number').simulate('blur', event);
            expect(Wrapper.state().error['number']).toBe(i18Data.invalidSupplierNumber);
        })
        test('it should not show error if the ipult value passes the validation', () => {        
            const event = { target: {id: 'number', value:'12345'}}
            Wrapper.find('.number').simulate('blur', event);
            expect(Wrapper.state().error['number']).toBe('');
        })
    })

    describe('Given the user submits the entered value', () => {
        test('it should validate the input name', () => {        
            const event = { preventDefault: jest.fn(), target: {id: 'name', value:'uni 7-'}}            
            Wrapper.find('.name').simulate('change', event);
            Wrapper.find('.search-icon').simulate('click', event);
            expect(Wrapper.state().error['name']).toBe(i18Data.invalidSupplierName);
        });
        test('it should validate the input number', () => {        
            const event = { preventDefault: jest.fn(), target: {id: 'number', value:'abc7-'}}            
            Wrapper.find('.number').simulate('change', event);
            Wrapper.find('.last').simulate('click', event);
            expect(Wrapper.state().error['number']).toBe(i18Data.invalidSupplierNumber);
        });
        test('it should pass the validation for the valid input', () => {        
            const event = { preventDefault: jest.fn(), target: {id: 'number', value:'12345'}}            
            Wrapper.find('.number').simulate('change', event);
            Wrapper.find('.search-icon').simulate('click', event);
            expect(Wrapper.state().error['number']).toBe('');
        });
        test('it should show error when there is no input value', () => {
            const event = { preventDefault: jest.fn(), target: {id: 'name', value:''}}            
            Wrapper.find('.name').simulate('change', event);
            Wrapper.find('.search-icon').simulate('click', event);
            expect(Wrapper.state().error['name']).toBe(i18Data.supplierNameError);
        });        

        test('should show error message if there is server error', async(done) => {        
            const event = {preventDefault: jest.fn(), target: {id: "name", value: "Supplier"}}            
            Wrapper.instance()._onChange(event)
            const error = "No partner found"
            httpClient.get.mockRejectedValue({message: error});
            Wrapper.find('.search-icon').simulate('click', event);
            setTimeout(() => {
                expect(Wrapper.state().error['name']).toBe(error);
                done();
            }, 0);
        });
        test('should show error message if it there is no data', async(done) => {
            const event = {preventDefault: jest.fn(), target: {id: "name", value: "Supplier"}}
            Wrapper.instance()._onChange(event)
            const error = 'No partner found';
            httpClient.get.mockResolvedValue({
                status: 404,
                message: error
            });
            Wrapper.find('.search-icon').simulate('click', event);
            setTimeout(() => {
                expect(Wrapper.state().error['name']).toBe(error)
                done();
            },0);
        });       

        test('should show error message if it fails to get the supplier by the number', async(done) => {
            const event = {preventDefault: jest.fn(), target: {id: "number", value: "12345"}}
            Wrapper.instance()._onChange(event)            
            httpClient.get.mockResolvedValue({suppliers:[]});

            Wrapper.find('.search-icon').simulate('click', event);
            setTimeout(() => {
                expect(Wrapper.state().error['number']).toBe(i18Data.supplierNotFound);
                done();
            },0);
        });
        test('should not show error message if it gets the suppliers', async(done) => {
            const event = {preventDefault: jest.fn(), target: {id: "name", value: "Supplier"}}
            const mockSearchMetaData = {
                "searchMetaData": {
                  "partnerStatus": {
                    "displayName": "Partner status"
                  },
                  "taxOrganisationType": {
                    "displayName": "Tax organisation type"
                  },
                  "partnerTaxRegime": {
                    "displayName": "Partner tax regime"
                  },
                  "siteInvoiceCurrency": {
                    "displayName": "Site invoice currency"
                  },
                  "productCategory": {
                    "displayName": "Product Cateogry"
                  },
                  "subGroup": {
                    "displayName": "Sub Group"
                  }
                }
              }
            Wrapper.setState({ searchMetaData: mockSearchMetaData });
            Wrapper.instance()._onChange(event)
            httpClient.get.mockResolvedValue({suppliers:[{name: "John", number: "123456"}], limit:10, offset:1});
            Wrapper.find('.search-icon').simulate('click', event);
            setTimeout(() => {
                expect(httpClient.get).toHaveBeenCalledWith('/tradingpartner/api/supplier?partnerName=Supplier&limit=10&offset=1');
                expect(Wrapper.state().error['name']).toBe('');
                done();
            },0);
        });
        test('should call the searchMetaData method when resolveCardData method is called', async(done) => {
            const mockSearchMetaData = {
                "searchMetaData": {
                  "partnerStatus": {
                    "displayName": "Partner status"
                  },
                  "taxOrganisationType": {
                    "displayName": "Tax organisation type"
                  },
                  "partnerTaxRegime": {
                    "displayName": "Partner tax regime"
                  },
                  "siteInvoiceCurrency": {
                    "displayName": "Site invoice currency"
                  },
                  "productCategory": {
                    "displayName": "Product Cateogry"
                  },
                  "subGroup": {
                    "displayName": "Sub Group"
                  }
                }
              }
            httpClient.get.mockResolvedValue(mockSearchMetaData);

            Wrapper.instance().resolveCardData()
            
            setTimeout(() => {
                expect(httpClient.get).toHaveBeenCalledWith('/tradingpartner/api/search-meta-data');
                expect(Wrapper.state().searchMetaData).toBe(mockSearchMetaData);
                done();
            },0);
        });
    });
    test('should show the results on page render, if the data is already present', async(done) => {
        const mockSearchResults = {
            "supplierResults": {
              "suppliers": [
                {
                  "entityType": "partner",
                  "uuid": "d682662a-7642-31d8-a7c1-e1dd1ddc7c47",
                  "number": "1443597821",
                  "name": "TEST98",
                  "taxOrganisationType": null
                }
              ],
              "totalCount": 1,
              "totalPages": 1,
              "offset": 1,
              "limit": 100
            },
            "searchedPartnerName": "TEST98"
          }
        const locationProp = {            
            state: mockSearchResults            
        }
        const Wrapper = shallow(<SearchSupplier history={mockHistory} location={locationProp}/>)
        
        setTimeout(() => {
            expect(httpClient.get).toHaveBeenCalledWith('/tradingpartner/api/search-meta-data');
            expect(Wrapper.state().searchResults).toBe(mockSearchResults.supplierResults.suppliers);
            done();
        },0);
    });

    test('should call searchSupplier method when getSearchResults method is called', async(done) => {
      const instance = Wrapper.instance();
      jest.spyOn(instance, 'searchSupplier');
      instance.getSearchResults()
      setTimeout(() => {
          expect(instance.searchSupplier).toHaveBeenCalled();
          done();
      },0);
  });
test('should show the error when _handleViewClick is called with error response', async(done) => {
  const instance = Wrapper.instance();
  let errorMessage = 'error'
  httpClient.post.mockResolvedValue({
    code: 404,
    message: errorMessage
});
  instance._handleViewClick()
  setTimeout(() => {
      expect(Wrapper.state().isError).toBe(true);
      expect(Wrapper.state().errorMessage).toBe(errorMessage);      
      done();
  },0);
});
});