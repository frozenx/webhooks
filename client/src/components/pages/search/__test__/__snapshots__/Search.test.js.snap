// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Given the Search component is rendered it should create a snapshot 1`] = `
<div
  className="base-components__RootElement-sc-150tvch-1 sc-gzVnrw gwVVlU"
>
  <div
    className="base-components__RootElement-sc-150tvch-1 sc-dnqmqq fzsTzb"
  >
    <h1>
      Find a Partner
    </h1>
    <h4
      className={false}
    >
      Enter one or more terms
    </h4>
  </div>
  <form
    onSubmit={[Function]}
  >
    <div
      className="base-components__RootElement-sc-150tvch-1 sc-bdVaJa dePZzZ"
    >
      <div
        className="base-components__RootElement-sc-150tvch-1 sc-bwzfXH dmfcSO"
      >
        <div
          className="base-components__RootElement-sc-150tvch-1 search-label styled__FormGroupContainer-wt4ift-1 hVjJdM"
          disabled={false}
        >
          <div
            className="base-components__BaseElement-sc-150tvch-0 styled__LinkContainer-wt4ift-3 iYRfeQ"
          />
          <div
            className="beans-form-group__children"
          />
        </div>
        <div
          className="base-components__RootElement-sc-150tvch-1 sc-bxivhb brAxYN"
        />
      </div>
    </div>
    <div
      className="base-components__RootElement-sc-150tvch-1 sc-bdVaJa dePZzZ"
    >
      <div
        className="base-components__RootElement-sc-150tvch-1 sc-htpNat ckasYZ"
      >
        <div
          className="base-components__RootElement-sc-150tvch-1 search-label styled__FormGroupContainer-wt4ift-1 hVjJdM"
          disabled={false}
        >
          <div
            className="base-components__BaseElement-sc-150tvch-0 styled__LinkContainer-wt4ift-3 iYRfeQ"
          />
          <div
            className="beans-form-group__children"
          />
        </div>
        <div
          className="base-components__RootElement-sc-150tvch-1 search-number sc-bxivhb brAxYN"
        >
          <div
            className="sc-gqjmRU jlBkwE"
          />
        </div>
      </div>
    </div>
  </form>
</div>
`;
